### ISSUE TITLE
Post-Cache Step Taking 150s in 2.7.2 Compared to 11-12s in 2.7.1

### ISSUE THREAD

**Issue by cwfitzgerald** [Swatinem/rust-cache#182]
Howdy, thanks for this awesome tool!

## Update

This is definitely an upstream problem with node20 - https://github.com/actions/toolkit/issues/1578

## Previous

Over at [wgpu](https://github.com/gfx-rs/wgpu/) we started getting 2.7.2 of rust-cache and our "post cache" jobs started taking ~15x whenever there was any uploading to do. I have a hunch this isn't a problem with rust-cache but with the core github cache action this is calling off to, but I wanted to start here to see if we came to the same conclusion.

If you look at the logs ([full log](https://github.com/gfx-rs/wgpu/commit/22e69191a25e6e75b9e5296cb6718aa6f41e2648/checks/20366834047/logs) and [post caching step.txt](https://github.com/Swatinem/rust-cache/files/13896110/34_Post.caching.txt)) I notice something suspicious with the timestamps:

```
2024-01-11T01:02:02.0366871Z ##[debug]Resource Url: https://acghubeus1.actions.githubusercontent.com/47bmaT25aAULNry7IX2PPcgYgwfrUqs3OBXgjQ0s8nspxqRVT2/_apis/artifactcache/caches/25694
2024-01-11T01:02:02.3201632Z Cache saved successfully
2024-01-11T01:04:12.6998275Z ##[debug]Node Action run completed with exit code 0
2024-01-11T01:04:12.7001105Z ##[debug]Finishing: Post caching
```

There's a 2 minute and 10 second disparity between the "cached saved successfully" log, and the debugging of github actions saying that node is finished. If I pin rust-cache to 2.7.1, this goes away.

My understanding is that "cached saved successfully" comes from the github cache code, not this code, so I am thinking it may be a problem there, but am curious what you think.

**Comment by Swatinem** [comment]
Thanks for reporting this and also providing a link to the upstream issue. Looks like there is a workaround, which I will integrate as well :-)

**Comment by Swatinem** [comment]
I published `2.7.3`, hope that helps. I added the same `process.exit()` workaround as was mentioned in the upstream issue.

Please reopen if this problem still persists.

**Comment by cwfitzgerald** [comment]
Seems to work well! About 100ms from the end of caching to the "end of job" message
